packages/terra-dev-site/src/layouts/secondary-navigation-layout/SecondaryNavigationLayout.jsx

```diff
@@ -0,0 +1,353 @@
+   *
+   * Providing a NavigationItem component as a direct child will result in a
+   * navigation entry being added to the SecondaryNavigationLayout side bar. Any
+   * non-NavigationItem children provided alongside NavigationItem children will
+   * **not** be rendered.
+   *
+   * If another layout is to be rendered within this layout, it is recommended
+   * to use the `renderLayout` prop instead. If the renderLayout prop is
+   * provided, children will be ignored, regardless of its contents.
+   */
+  children: PropTypes.node,
+};
+
+function mapChildItem(item) {
+  return {
+    text: item.label,
+    name: item.label,
+    path: item.key,
+    childItems: item.childItems ? item.childItems.map(mapChildItem) : undefined,
+  };
+}
+
+const SecondaryNavigationLayout = ({
+  id,
+  label,
+  activeNavigationKey,
+  children,
+  onSelectNavigationItem,
+  renderPage,
+  renderLayout,
+  renderNavigationFallback,
+}) => {
+  const activeBreakpoint = React.useContext(ActiveBreakpointContext);
+  const parentLayoutActions = React.useContext(LayoutActionsContext);
+  const theme = React.useContext(ThemeContext);
+
+  const pageContainerRef = React.useRef();
+  const sideNavBodyRef = React.useRef();
+  const sideNavPanelRef = React.useRef();
+  const sideNavOverlayRef = React.useRef();
+
+  const [contentElementRef, pageContainerPortalsRef] = usePortalManager(activeNavigationKey, () => {
+    deferExecution(() => {
+      document.body.focus();
+    });
+  });
+
+  const [sideNavOverlayIsVisible, setSideNavOverlayIsVisible] = React.useState(false);
+
+  function getNavigationItems(childComponents) {
+    return React.Children.toArray(childComponents).reduce((accumulator, child) => {
+      const items = [...accumulator];
+
+      if (child.type === NavigationItem) {
+        items.push(child);
+      } else if (child.type === SecondaryNavigationGroup) {
+        const groupItems = getNavigationItems(child.props.children);
+        if (groupItems) {
+          items.push(...groupItems);
+        }
+      }
+
+      return items;
+    }, []);
+  }
+
+  const navigationItems = getNavigationItems(children);
+  const hasActiveNavigationItem = !!navigationItems.find(item => item.props.navigationKey === activeNavigationKey);
+
+  const hasSidebar = !!navigationItems.length;
+  const hasOverlaySidebar = sideNavOverlayBreakpoints.indexOf(activeBreakpoint) !== -1;
+  const sideNavIsVisible = hasSidebar && (sideNavOverlayIsVisible || sideNavOverlayBreakpoints.indexOf(activeBreakpoint) === -1);
+
+  const layoutActionsContextValue = React.useMemo(() => {
+    let newStartActions = parentLayoutActions.startActions;
+
+    if (hasSidebar && hasOverlaySidebar) {
+      newStartActions = [...newStartActions, {
+        key: 'secondary-navigation-layout-toggle-navigation-panel',
+        label: `Toggle Navigation Panel ${sideNavOverlayIsVisible ? 'Closed' : 'Open'}`, // TODO intl and verify a11y
```

Comment:
Translate this?

Reply:
Normally yes, but since this component is specific to dev site and none of our documentation is translated I think it's fine if we leave this un translated until it makes it into terra application v2